mmv1/products/spanner/api.yaml

```diff
@@ -102,8 +102,19 @@ objects:
         required: true
       - !ruby/object:Api::Type::Integer
         name: 'nodeCount'
-        description: 'The number of nodes allocated to this instance.'
-        default_value: 1
+        description: |
+          The number of nodes allocated to this instance. At most one of either node_count or processing_units can be present in terraforn. 
+        exactly_one_of:
+          - num_names
```

Comment:
```suggestion
          - num_nodes
```

Reply:
I tried that and it doesn't work. Trying conflicts_with 

Pushed a commit with correct field name

brb

Reply:
mmv1 is buggy.

```
I, [2021-07-23T22:17:44.793294 #95375]  INFO -- : products/pubsub: Not specified, skipping generation
bundler: failed to load command: compiler (compiler)
RuntimeError: num_nodes does not exist
  /Users/maha4472/Desktop/Git/magic-modules/mmv1/api/type.rb:218:in `block in check_conflicts'
  /Users/maha4472/Desktop/Git/magic-modules/mmv1/api/type.rb:217:in `each'
  /Users/maha4472/Desktop/Git/magic-modules/mmv1/api/type.rb:217:in `check_conflicts'
  /Users/maha4472/Desktop/Git/magic-modules/mmv1/api/type.rb:136:in `validate'
  /Users/maha4472/Desktop/Git/magic-modules/mmv1/google/yaml_validator.rb:121:in `check_property_value'
  /Users/maha4472/Desktop/Git/magic-modules/mmv1/google/yaml_validator.rb:75:in `block in check'
  /Users/maha4472/Desktop/Git/magic-modules/mmv1/google/yaml_validator.rb:74:in `each'
  /Users/maha4472/Desktop/Git/magic-modules/mmv1/google/yaml_validator.rb:74:in `each_with_index'
  /Users/maha4472/Desktop/Git/magic-modules/mmv1/google/yaml_validator.rb:74:in `check'
  /Users/maha4472/Desktop/Git/magic-modules/mmv1/api/resource.rb:173:in `validate'
  /Users/maha4472/Desktop/Git/magic-modules/mmv1/google/yaml_validator.rb:121:in `check_property_value'
  /Users/maha4472/Desktop/Git/magic-modules/mmv1/google/yaml_validator.rb:75:in `block in check'
  /Users/maha4472/Desktop/Git/magic-modules/mmv1/google/yaml_validator.rb:74:in `each'
  /Users/maha4472/Desktop/Git/magic-modules/mmv1/google/yaml_validator.rb:74:in `each_with_index'
  /Users/maha4472/Desktop/Git/magic-modules/mmv1/google/yaml_validator.rb:74:in `check'
  /Users/maha4472/Desktop/Git/magic-modules/mmv1/api/product.rb:59:in `validate'
  compiler:178:in `block in <top (required)>'
  compiler:144:in `each'
  compiler:144:in `<top (required)>'
```

Regardless a schema like this doesn't work
```
			"num_nodes": {
				Type:          schema.TypeInt,
				Optional:      true,
				Description:   `The number of nodes allocated to this instance. At most one of either node_count or processing_units can be present in terraforn.`,
				Default:       1,
				ConflictsWith: []string{"processing_units"},
			},
			"processing_units": {
				Type:     schema.TypeInt,
				Optional: true,
				Description: `The number of processing units allocated to this instance. At most one of processing_units 
or node_count can be present in terraform.`,
				ConflictsWith: []string{"num_nodes"},
			},
```

Causes this error:

```
021/07/23 23:20:40 [DEBUG] Retry Transport: starting RoundTrip retry loop
2021/07/23 23:20:40 [DEBUG] Retry Transport: request attempt 0
2021/07/23 23:20:40 [DEBUG] Google API Request Details:
---[ REQUEST ]---------------------------------------
POST /v1/projects/REDACTED/instances?alt=json HTTP/1.1
Host: spanner.googleapis.com
User-Agent: Terraform/0.14.11 (+https://www.terraform.io) Terraform-Plugin-SDK/2.5.0 terraform-provider-google/acc
Content-Length: 231
Content-Type: application/json
Accept-Encoding: gzip

{
 "instance": {
  "config": "projects/REDACTED/instanceConfigs/regional-us-central1",
  "displayName": "Test Spanner Instance",
  "labels": {
   "foo": "bar"
  },
  "nodeCount": 1,
  "processingUnits": 200
 },
 "instanceId": "tfgen-spanid-20210723222040904"
}

-----------------------------------------------------
2021/07/23 23:20:41 [DEBUG] Google API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 400 Bad Request
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-T051=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Fri, 23 Jul 2021 22:20:41 GMT
Server: ESF
Server-Timing: gfet4t7; dur=446
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

{
  "error": {
    "code": 400,
    "message": "Invalid CreateInstance request. At most one of node_count or processing_units may be specified.",
    "status": "INVALID_ARGUMENT"
  }
}

-----------------------------------------------------
2021/07/23 23:20:41 [DEBUG] Retry Transport: Stopping retries, last request failed with non-retryable error: googleapi: got HTTP response code 400 with body: HTTP/2.0 400 Bad Request
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-T051=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Fri, 23 Jul 2021 22:20:41 GMT
Server: ESF
Server-Timing: gfet4t7; dur=446
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

{
  "error": {
    "code": 400,
    "message": "Invalid CreateInstance request. At most one of node_count or processing_units may be specified.",
    "status": "INVALID_ARGUMENT"
  }
}
2021/07/23 23:20:41 [DEBUG] Retry Transport: Returning after 1 attempts
2021/07/23 23:20:41 [WARN] Got error running Terraform: exit status 1

Error: Error creating Instance: googleapi: Error 400: Invalid CreateInstance request. At most one of node_count or processing_units may be specified.

  on terraform_plugin_test.tf line 2, in resource "google_spanner_instance" "example":
   2: resource "google_spanner_instance" "example" {


    provider_test.go:268: Step 1/2 error: Error running apply: exit status 1
        
        Error: Error creating Instance: googleapi: Error 400: Invalid CreateInstance request. At most one of node_count or processing_units may be specified.
        
          on terraform_plugin_test.tf line 2, in resource "google_spanner_instance" "example":
           2: resource "google_spanner_instance" "example" {
        
        
--- FAIL: TestAccSpannerInstance_spannerInstanceProcessingUnitsExample (18.62s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-google/google	21.535s
FAIL

```

Removing the default value works but it is a breaking change

Reply:
Also, you can't avoid ExactlyOneOf

```
2021/07/23 23:25:43 [DEBUG] Google API Request Details:
---[ REQUEST ]---------------------------------------
POST /v1/projects/REDSACTED/instances?alt=json HTTP/1.1
Host: spanner.googleapis.com
User-Agent: Terraform/0.14.11 (+https://www.terraform.io) Terraform-Plugin-SDK/2.5.0 terraform-provider-google/acc
Content-Length: 195
Content-Type: application/json
Accept-Encoding: gzip

{
 "instance": {
  "config": "projects/REDACTED/instanceConfigs/regional-us-central1",
  "displayName": "Test Spanner Instance",
  "labels": {
   "foo": "bar"
  }
 },
 "instanceId": "tfgen-spanid-20210723222543515"
}

-----------------------------------------------------
2021/07/23 23:25:44 [DEBUG] Google API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 400 Bad Request
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-T051=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Fri, 23 Jul 2021 22:25:44 GMT
Server: ESF
Server-Timing: gfet4t7; dur=406
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

{
  "error": {
    "code": 400,
    "message": "Invalid CreateInstance request.",
    "status": "INVALID_ARGUMENT",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.BadRequest",
        "fieldViolations": [
          {
            "field": "instance.node_count",
            "description": "Node count must be a positive number."
          }
        ]
      }
    ]
  }
}

-
```